From d08073794a71c5bf2ad1df52799d359f55da0680 Mon Sep 17 00:00:00 2001
From: "A. Wilcox" <AWilcox@Wilcox-Tech.com>
Date: Wed, 10 Jul 2019 19:44:48 -0500
Subject: [PATCH] add: create_virtual_package: dup ver string

ver is a stack-allocated variable.  Its scope ends when the function does.
This means that the version atom is no longer valid after the return of
create_virtual_package.
---
 src/add.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/add.c b/src/add.c
index 4c285f7..7e15f10 100644
--- a/src/add.c
+++ b/src/add.c
@@ -97,7 +97,7 @@ static struct apk_package *create_virtual_package(struct apk_database *db, struc
 	if (virtpkg == NULL) return 0;
 
 	virtpkg->name = name;
-	virtpkg->version = apk_blob_atomize(APK_BLOB_STR(ver));
+	virtpkg->version = apk_blob_atomize_dup(APK_BLOB_STR(ver));
 	virtpkg->description = strdup("virtual meta package");
 	virtpkg->arch = apk_blob_atomize(APK_BLOB_STR("noarch"));
 
-- 
2.22.0
